crates/README.md

```diff
@@ -0,0 +1,147 @@
+
+# Roc Internals
+
+Roc has different rust crates for various binaries and libraries. These are described below to help give you an overview. If you see something here that is out of date, please correct it or discuss it in the [Zulip chat](https://roc.zulipchat.com/).
+
+Use `cargo doc` to generate docs; e.g. ```cargo doc --package roc_ast --open```.
+
+## `ast/` - `roc_ast`
+
+Code to represent the [Abstract Syntax Tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) as used by the editor.
+In contrast to the compiler, the types in this AST do not keep track of the location of the matching code in the source file.
+
+
+## `cli/` - `roc_cli`
+
+The `roc` binary that brings together all functionality in the Roc toolset.
+
+## `cli_utils/` - `cli_utils`
+
+Provides shared code for cli tests and benchmarks.
+
+## `code_markup/` - `roc_code_markup`
+
+Roc's very own [markup language](https://en.wikipedia.org/wiki/Markup_language). This library is used by the editor to display good-looking roc code. 
+
+## `compile/`
+
+Compiles `.roc` files and combines them with their platform into an executable binary. See [compiler/README.md](./compiler/README.md) for more information.
+
+TODO explain what "compiler frontend" is
+TODO explain what "compiler backend" is
```

Comment:
Maybe worth skipping for now, this can be elaborated in a follow-up. Generally the compiler is the largest piece of the Roc project, a good introduction would be useful here; there is one we'd probably want to combine with this document in compiler/README.md. But, that doesn't need to be done in this patch.